Devil agent

lordpidey opened the pull request:
:cl:
add: Added Devil agent gamemode, where multiple devils are each trying to buy more souls than the next in line.
add: If you've already sold your soul, you can sell it again to a different devil.  You can even go back and forth for INFINITE POWER.
/:cl:

This adds a new gamemode.  Devil Agent.  It works much like Double Agent, except devils instead of traitors, and instead of killing, the devil simply has to control more souls than the enemy.

Wew, I'm ready for this to be reviewed.

Also, whether this is included in the rotation depends on the config settings.  By default, it will be disabled.

lordpidey committed:
Starts work on "devil agent" gamemode.

lordpidey committed:
Merge remote-tracking branch 'tgstation/master' into multi_devil

lordpidey committed:
Minor refactors to older devil objectives.
Further implementations of new devil objectives.

lordpidey committed:
More work on the devil gamemode.

lordpidey committed:
Fixes compile errors.

lordpidey committed:
You may sign a contract, so long as your soul isn't owned by THAT devil.

lordpidey committed:
outsell objectives now show correct completion.

lordpidey committed:
fixes compile error

lordpidey committed:
fixes compile error

GunHog commented:
I hope this pans out for you! ^_^

jughu commented:
N-no more devils m-moar angels pls

Jalleo commented on code/game/gamemodes/devil/devil agent/devil_agent.dm:
```diff
@@ -0,0 +1,42 @@
+
+/datum/game_mode/devil/devil_agents/announce()
+	world << "<B>The current game mode is - Devil Agents!</B>"
+	world << "<B>There are several devils onboard the station, trying to each sell more souls than the other.</B>"
```
Abuy buy buy. Wait did you hear i wanted to sell souls I want to buy your soul

Bawhoppen commented:
gamemode code freeze

lordpidey committed:
small text fixes.

lordpidey committed:
Fixes remaining bugs.

lordpidey committed:
Merge branch 'master' into multi_devil

lordpidey committed:
Converts to new announce system.

lordpidey committed:
Devils spawned through events and created via traitor panel now start with objectives.

Robustin commented:
Last I checked Devils were utterly shit. Why the fuck would a mode involving more of them be a good thing?

"Hey assistant, u want all-access?"
"Sure"
"Ok"
~Soul Acquired~
~Greytide Engaged~

KorPhaeron commented:
Not sure our server is high enough RP for this. Devil has lots of fluffy limits you're supposed to follow/rules/lore whatever, but everyone I saw was just running around screaming "HEY BRO WANT TO SELL YOUR SOUL?"

BananaSenor commented:
Joined late into such a round as lawyer. Hit the evil Captain with his contract, saw the message he is back on the team, yet he could not be revived after he died a few seconds later. Sold my own soul, deconverted myself, but kept the powers as well. Stuff seems to be broken.

lordpidey commented:
@BananaSenor You keep your boons when you regain (or resell) your soul, that is intentional.  Devils are supposed to try to prevent this, it's one of the tasks they have.  However, being unrevivable after regaining your soul is a bug.

@KorPhaeron There should only be three RP limits a devil has.  Don't hurt other devils.  A random obligation, and a random forbiddance.  I'd be okay with dropping it to the don't hurt devils and one random obligation OR random forbiddance.  The Bane and Banishment traits of a devil are mechanically enforced.

lordpidey committed:
Tones down the number of devils, because holyshit, that was too many.

lordpidey committed:
Returning someone's soul now makes them revivable.

lordpidey committed:
Fixes the "Buy the soul of X" objective coming out as free objective.

lordpidey committed:
true devils now dense

lordpidey committed:
Amount of soulpower lost per death is now a define, and set to 2.

lordpidey committed:
Fixes compile error, and actually fixes "free objective"

lordpidey committed:
Neatens up end of round devil/sintouched announcements.

lordpidey committed:
Adds progress bar to infernal jaunt.
Infernal jaunt may now jaunt in in shuttles, regardless of if there is a nearby victim.

lordpidey committed:
Jaunting in on a shuttle now actually works.
Jaunting in now gives a TINY amount of feedback.

lordpidey committed:
Fixes compile error

lordpidey committed:
Check antagonist panel's check devil info now only shows information on devils you request, rather than all.

lordpidey commented:
Aright, I've finished the feedback I've recieved from the first test.  There were a lot of problems, and given how hectic that one round was, there will likely need to be another test to look for more problems.

lordpidey committed:
Devils now know not only the truename, but the mortal name of their targets.

optimumtact commented:
So this is still in progress?

lordpidey commented:
I recieved no feedback from the 2nd test, except one person was unable to move upon ascending to third stage devil (Lanky devil), and I can't figure out why, I also can't seem to reproduce the bug.  I'd like to track that bug down, but as I cannot reproduce it, I'm having a hard time.

GeneralGoose commented:
sold soul = cluwnban 2 years

lordpidey commented:
There was already a discussion about penalties for selling your soul.  Ultimately, it was decided that doing nothing would be best.

GunHog commented:
> There was already a discussion about penalties for selling your soul. Ultimately, it was decided that doing nothing would be best.

If the Devil dies, he can steal your body, forcibly ghosting you. How about that?

lordpidey commented:
@GunHog, that is a very interesting idea, but outside of the scope of this PR.  I'm not sure if I like it, as legends about selling your soul only have consequences after death.

PKPenguin321 commented:
Who cares about irl legends
